server/middleware.go

```diff
@@ -0,0 +1,66 @@
+package server
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/Layr-Labs/eigenda-proxy/commitments"
+	"github.com/Layr-Labs/eigenda-proxy/metrics"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+// withMetrics is a middleware that records metrics for the route path.
+func withMetrics(
+	handleFn func(http.ResponseWriter, *http.Request) error,
+	m metrics.Metricer,
+	mode commitments.CommitmentMode,
+) func(http.ResponseWriter, *http.Request) error {
+	return func(w http.ResponseWriter, r *http.Request) error {
+		recordDur := m.RecordRPCServerRequest(r.Method)
+
+		err := handleFn(w, r)
+		if err != nil {
+			var metaErr MetaError
+			if errors.As(err, &metaErr) {
+				recordDur(w.Header().Get("status"), string(metaErr.Meta.Mode), string(metaErr.Meta.CertVersion))
+			} else {
+				recordDur(w.Header().Get("status"), string("NoCommitmentMode"), string("NoCertVersion"))
```

Comment:
what if we instead move these strings to introduce some `unknown` type for commitment mode and cert version enums?

Reply:
[b91ab13](https://github.com/Layr-Labs/eigenda-proxy/pull/185/commits/b91ab13c275626a51e78649c5f07a253dd2954fe)
Changed them both to just say "unknown" instead, but prefer not to introduce a type for this simple case. If it was used elsewhere I'd be fine with it, but here I think its clearer to just have the string written, instead of seeing a variable and then needing to click-follow in the IDE to see what string is going to appear in the actual prometheus metric.